### DISCUSSION TITLE
Trouble cloning my organization's repository

### DISCUSSION THREAD

**Discussion by amart2002** [community/community#58154] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hello,

I am having trouble cloning a github repository, I have the right access to the repository, I created the SSH key and pasted it in the github and obtained a new SSH key from the repository and I am at the part where i use the command "git clone " but I keep getting this same error:

fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

1. The steps I followed are:
2. In terminal, run: ssh-keygen -t ed25519 -C "[your_email@example.com]
3. Press enter three times (I think) until process appears complete
4. In terminal, run: cat ~/.ssh/id_rsa.pub (may have to tab complete this to get it to find file)
5. Copy the whole printed statement that starts with ssh
6. Go to github in browser, click your avatar in top right and navigate to settings
7. In sidebar, click SSH and GPG keys, then click the green New SSH key button
8. Give it a title and copy the whole statement from step 4 into the key box
9. In terminal, run: eval "$(ssh-agent -s)"
10. In https://github.com/<organization github>, click the green Code button, click ssh, and then copy the given ssh key
11. In terminal, cd into the directory that you want the code to be in
12. In terminal, run: git clone (paste ssh key here) ERROR

I am stuck in step 12 since I keep getting the same error.
"""
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists. """

How can I go about this issue, if I know I have the right access, and I have done the steps to make a new SSH key and retrieve the one from the repository I want to clone.

**Comment by airtower-luna** [comment, marked as answer]
> In terminal, run: git clone (paste ssh key here) ERROR

This part is wrong. You need to give `git clone` the SSH URL for your repository, not your key. If the key is encrypted and not loaded into `ssh-agent` yet you'll be asked for the passphrase when Git tries to use it.

**Comment by Yohannes022** [comment]
![Web capture_30-6-2023_214850_intranet alxswe com](https://github.com/community/community/assets/99507599/7b51fb18-78bb-4c93-b159-10f7d3d3cea7)
 How can handle this thing

**Reply by airtower-luna** [reply]
* You need to use your exact username.
* When Git asks for your password, you need to provide a personal access token (PAT). When [creating the PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files). Usually you should store the token in your password manager.

That has nothing to do with the question above though, you're using HTTPS, not SSH. :wink:
